transaction/_manager.py

```diff
@@ -174,13 +175,15 @@ def run(self, func=None, tries=3):
         doc = func.__doc__
         if name != '_':
             if doc:
-                doc = name + '\n\n' + doc
+                doc = name + u'\n\n' + doc
             else:
                 doc = name
 
         for i in range(1, tries + 1):
             txn = self.begin()
             if doc:
+                if not isinstance(doc, text_type):
+                    doc = doc.decode('ascii')
```

Comment:
I do not know of a way to get the file encoding after the fact. That information is explicitly thrown away by PEP263:

>  compile it, creating Unicode objects from the given Unicode data and creating string objects from the Unicode literal data by first reencoding the UTF-8 data into 8-bit string data using the given file encoding

So UTF8 was just a guess about what's probably most common. And if that doesn't work, well, latin-1 is non-lossy and the data can be reconstructed later. Or we could do what pyramid_tm does, which is to catch the decoding error and replace it with a specific constant string ("Unable to decode path as unicode" in its case).